confirm.c

```diff
@@ -763,16 +766,17 @@ confirm_query_create4(const struct sockaddr_storage *dst, uint8_t ttl,
 }
 
 struct confirm_query *
-confirm_query_create6(const struct sockaddr_storage *dst, uint8_t ttl,
-		uint8_t traffic_class, uint32_t flow_label,
-		uint16_t icmpid, uint8_t flowid,
-		confirm_query_cb cb)
+confirm_query_create6_icmp(const struct sockaddr_storage *dst, uint8_t ttl,
+		uint8_t traffic_class, uint32_t flow_label, uint16_t icmpid,
+		uint8_t flowid, confirm_query_cb cb)
 {
-	struct confirm_query *query;
-
-	query = malloc(sizeof(struct confirm_query));
+	struct confirm_query *query = malloc(sizeof(struct confirm_query));
 	if(!query) logea(__FILE__, __LINE__, NULL);
 
+	if(flowid > CONFIRM_MAX_FLOWID) {
+		logd(LOG_WARN, "%s,%d: flowid > 127!\n", __FILE__, __LINE__);
```

Comment:
Yeah, this is what I am not understanding. In the version I have we are deciding where to send a TCP or ICMP packet based on the `query->icmpid` field [1], which is probably not what you have today.

https://github.com/cunha/netlib/blob/884b4228d738c589604857b109685d2b72d3badd/confirm.c#L603